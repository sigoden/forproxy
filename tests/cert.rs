mod common;

use crate::common::{build_proxy_client, start_http_server, start_proxy};
use proxyfor::server::CERT_PREFIX;

use anyhow::Result;

#[tokio::test]
async fn test_cert_page() -> Result<()> {
    let (proxy_addr, stop_proxy) = start_proxy(false).await?;
    let (_server_addr, stop_server) = start_http_server().await?;

    let client = build_proxy_client(&proxy_addr.to_string())?;

    let res = client.get(CERT_PREFIX).send().await?;

    assert_eq!(res.status(), 200);
    insta::assert_snapshot!(res.text().await?);

    let _ = stop_server.send(());
    let _ = stop_proxy.send(());
    Ok(())
}

#[tokio::test]
async fn test_cert_cer() -> Result<()> {
    let (proxy_addr, stop_proxy) = start_proxy(false).await?;
    let (_server_addr, stop_server) = start_http_server().await?;

    let client = build_proxy_client(&proxy_addr.to_string())?;

    let res = client
        .get(format!("{}proxyfor-ca-cert.cer", CERT_PREFIX))
        .send()
        .await?;

    assert_eq!(res.status(), 200);
    insta::assert_snapshot!(res.text().await?);

    let _ = stop_server.send(());
    let _ = stop_proxy.send(());
    Ok(())
}

#[tokio::test]
async fn test_cert_pem() -> Result<()> {
    let (proxy_addr, stop_proxy) = start_proxy(false).await?;
    let (_server_addr, stop_server) = start_http_server().await?;

    let client = build_proxy_client(&proxy_addr.to_string())?;

    let res = client
        .get(format!("{}proxyfor-ca-cert.pem", CERT_PREFIX))
        .send()
        .await?;

    assert_eq!(res.status(), 200);
    insta::assert_snapshot!(res.text().await?);

    let _ = stop_server.send(());
    let _ = stop_proxy.send(());
    Ok(())
}
