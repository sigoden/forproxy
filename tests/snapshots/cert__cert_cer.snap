---
source: tests/cert.rs
expression: text
---
-----BEGIN CERTIFICATE-----
MIIFLTCCAxWgAwIBAgIUHcH7Cvpim0xvgDgpS0aKrXNAowwwDQYJKoZIhvcNAQEL
BQAwJjERMA8GA1UEAwwIcHJveHlmb3IxETAPBgNVBAoMCHByb3h5Zm9yMB4XDTI0
MDMxNTA2NDYxMFoXDTM0MDMxMzA2NDYxMFowJjERMA8GA1UEAwwIcHJveHlmb3Ix
ETAPBgNVBAoMCHByb3h5Zm9yMIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKC
AgEAzkCw1/tYhWOD8pPDGBtcq4Zp8nzXqyLdkCbiPgtXCct/FbYWzrv7ph1V5eyX
cOgP5YucYfmCtU0R6z9SX9kKTlMwk4Or4XNdWAJ0Db5qr55MKcTM7Zewr2spS/4d
GNXSGejTIR9NSByLYWJ/HnrgqqTxSNFGyeFlmCYYTYn0E68gxunfHd7X+BQS2VGB
+cWO4oc6qisN29q16N6ab54UFIfyxoCQawu3siiw9SbsfDMqkxysXpH24GUcoDMM
K/ZMzRBFbbL0bjpzlt/LFErP+WdVdgFyMgu36f3vsg8DJcYjIAC0zGdjVwVt61kx
eQoijNytJ2zwjhFTuQ+g3eEDx3tQXQ+/Gf/9IsmyYro27JEZ5pO6/FKtf7gPkcAx
d2WAMZXXm+OQr6gmRYkddOsaJcNsceUqjx6E0ZTm6tAXh8A0JIERVLZ9eLhm+98p
lzjair9L2ecVhFwV9dryrkmjfnbkMfVxqeq5R5U8uZRFJKLCZc/4qAossUA7zdi2
R7xCagoR4NIYZJ1w+yxemOVzT1BtoFJX76yl+KOFgh8niRJAKmU9FiLaI1akoRa5
csM2HxZjTBHqcw2iTreAcWtVomR3rGWlb9GgDVVoeie9D6PDhb05mv0csD+R8aOT
vx9CKJJBENXuqvu+1pxkEOQ4Ol/phjCinZevUb1UJ4C79GcCAwEAAaNTMFEwHQYD
VR0OBBYEFPlP/DXoQehmsUfMQHKivd7hfdgGMB8GA1UdIwQYMBaAFPlP/DXoQehm
sUfMQHKivd7hfdgGMA8GA1UdEwEB/wQFMAMBAf8wDQYJKoZIhvcNAQELBQADggIB
AHFxf/+b8KeewPF2B8UByGPqrnDW9ixEr2ol/TjyL45maWiE1U+UDA3JjkSbtMd2
bHgcQQqyyYaJDOfvkrd+S5flwkmLl+h5QH9hjxfu+j05/yBOt5u4TbPjZLW+Id+1
1WxoRaGLAmvh7l9IS+xu/UBZWcm4xjsyygduvyHqdB5vjZXdJ9egB371ebRExKM6
aGehG4c6+caDC0uG+A23sBV+DZ2o9j39I59TTv7KmiUVEvexxOzFxgqVLQrUnkyi
I1wxYHD9ytdlh+X6c5mM4L/QDFYwKGMUyqmwJ8yrrSWLzXKIMFljkfPczjKKr2mn
SvVFGIOiBrGAY5oSexgKf80JCQ5SX9Do+55jjYOVe7i2H5vi9X0og6np1Zb42Pt6
pwlU47HUlF9hNqR516JFCbVDNByZzXN15uUWa98/VGZNFTgwya5uTb9tIWwgKTfN
BsmqaS3JiXggAeoJ+TkhLHeesXp+uyRT4G0JgbMvR76+3vF9BaTRqY67Cujz7zZG
J3s5LRBiV9Ey1kjFfAsYvy/3HJzrPaUKSUFVoqPPlv9fdtQMXDFNYHre+DRnV1tY
a1GqF6c0VYooBiYF21mX5lXv4C67h0rSZluvjtTTOXznYEoaUH/G2gPe2Ke+Yo7s
YJnYoagsdgp8wgVpTbhRtJbxMRUhM617oQIm6M24squ4
-----END CERTIFICATE-----
